aggregate.go

```diff
@@ -1,69 +1,85 @@
+// domain event messages.
 //
-// Each command message is routed to a single aggregate instance, represented by
+// Each command message targets a single aggregate instance, represented by
 // the AggregateRoot interface.
 type AggregateMessageHandler interface {
 	// New returns a new aggregate instance.
 	New() AggregateRoot
 
-	// RouteCommand determines how to route a command message to this handler,
-	// based on its type.
+	// RouteCommand indicates whether a specific type or instance of a command
+	// message should be routed to this handler.
 	//
-	// If ok is true, m is routed to this handler, within the context of the
-	// aggregate instance nominated by id. id must not be empty.
+	// If p is false, then m is a command message that has been sent to the
+	// application. If m should be routed to this handler the implementation sets
+	// ok to true, and id to a non-empty value indicating the ID of the aggregate
+	// instance that the command targets.
 	//
-	// If p is true, a "routing probe" is being performed, in which case m is a
-	// zero-value and id is ignored. ok should be true if this handler should
-	// receive messages of m's type.
+	// If p is true, then the engine is performing a "routing probe". In this case
+	// m is a non-nil, zero-value message. The implementation sets ok to true if
+	// messages of this type should be routed to this message handler when they
+	// occur. The id output parameter is unused.
 	RouteCommand(m Message, p bool) (id string, ok bool)
 
-	// HandleCommand handles a command message within the scope of a specific
-	// aggregate instance.
+	// HandleCommand handles a command message that has been routed to this
+	// handler.
 	//
-	// It inspects the state of the instance returned by s.Instance() to determine
-	// what state changes, if any, the command should produce.
+	// Handling a command involves inspecting the state of the command's target
+	// aggregate instance to determine what changes, if any, should occur. Each
+	// change is indicated by recording an event message.
 	//
-	// Any such change is indicated by calling s.RecordEvent() with an event
-	// message that represents that state change. The instance MUST NOT be modified
-	// directly by this method. Instead, the modifications must be made by the
-	// instance's ApplyEvent() method, which is called for each recorded event.
+	// s provides access to the operations available within the scope of handling
+	// m, such as loading the targetted instance and recording event messages.
 	//
-	// If m is of an unsupported message type, it must panic with an
-	// UnexpectedMessage error.
+	// This method must not manipulate the targetted instance directly. Any such
+	// manipulations must be applied by the instance's ApplyEvent() method,
+	// which is called for each recorded event message.
+	//
+	// If m was not expected by the handler, as per the routes determined by calls
+	// to RouteCommand(), it must panic with an UnexpectedMessage error.
```

Comment:
Should **error** be removed from this now that `UnexpectedMessage` is not an actual `error`?